vispy/visuals/tube.py

```diff
@@ -0,0 +1,171 @@
+        to color only by the `color` argument. Defaults to None.
+    shading : str
+        Same as for the `MeshVisual` class. Defaults to 'smooth'.
+    vertex_colors: ndarray or None
+        Same as for the `MeshVisual` class.
+    face_colors: ndarray or None
+        Same as for the `MeshVisual` class.
+    color : Color
+        The `Color` to use when drawing the tube. Defaults to (1, 0, 1, 1).
+    mode : str
+        Same as for the `MeshVisual` class. Defaults to 'triangles'.
+    """
+    def __init__(self, points, radius=1.0, tube_points=8,
+                 colors=None,
+                 closed=False,
+                 shading='smooth',
+                 vertex_colors=None,
+                 face_colors=None,
+                 color=(1, 0, 1, 1),
+                 mode='triangles'):
+
+        points = np.array(points)
+
+        tangents, normals, binormals = frenet_frames(points, closed)
+
+        segments = len(points) - 1
+
+        # get the positions of each vertex
+        grid = np.zeros((len(points), tube_points, 3))
+        for i in range(len(points)):
+            pos = points[i]
+            normal = normals[i]
+            binormal = binormals[i]
+
+            # Add a vertex for each point on the circle
+            v = np.arange(tube_points, dtype=np.float) / tube_points * 2 * np.pi
+            cx = -1. * radius * np.cos(v)
+            cy = radius * np.sin(v)
+            grid[i] = (pos + cx[:, np.newaxis]*normal +
+                       cy[:, np.newaxis]*binormal)
+
+        # construct the mesh
+        indices = []
+        for i in range(segments):
+            for j in range(tube_points):
+                ip = (i+1) % segments if closed else i+1
+                jp = (j+1) % tube_points
+
+                index_a = i*tube_points + j
+                index_b = ip*tube_points + j
+                index_c = ip*tube_points + jp
+                index_d = i*tube_points + jp
+
+                indices.append([index_a, index_b, index_d])
+                indices.append([index_b, index_c, index_d])
+
+        vertices = grid.reshape(grid.shape[0]*grid.shape[1], 3)
+
+        if colors is not None and vertex_colors is None:
+            vertex_colors = np.zeros(vertices.shape, dtype=np.float32)
+            vertex_colors[:, 0] = np.repeat(colors[:, 0], tube_points)
+            vertex_colors[:, 1] = np.repeat(colors[:, 1], tube_points)
+            vertex_colors[:, 2] = np.repeat(colors[:, 2], tube_points)
+
+        indices = np.array(indices, dtype=np.uint32)
+
+        MeshVisual.__init__(self, vertices, indices,
+                            vertex_colors=vertex_colors,
+                            face_colors=face_colors,
+                            color=color,
+                            shading=shading,
+                            mode=mode)
+                
+
+    def draw(self, transforms):
+        MeshVisual.draw(self, transforms)
+
+
+def frenet_frames(points, closed):
```

Comment:
This should probably be private, so please add a leading underscore to the function name.